ci/generate-workflows.py

```diff
@@ -54,19 +58,22 @@ def __init__(self, name, action, params=None, env=None, condition=None):
         self.env = env
         self.condition = condition
 
-    def render(self, f, env):
-        f.write(f"    - name: {yv(self.name)}\n")
-        f.write(f"      uses: {self.action}\n")
+    def render(self, f, env, depth=0):
+        indent = "  " * depth
+        f.write(f"{indent}- name: {yv(self.name)}\n")
+        f.write(f"{indent}  uses: {self.action}\n")
         if self.condition:
-            f.write(f"      if: {self.condition}\n")
+            f.write(f"{indent}  if: {self.condition}\n")
         if self.params:
-            f.write("      with:\n")
+            f.write(f"{indent}  with:\n")
             for k, v in self.params.items():
-                f.write(f"         {k}: {yv(v)}\n")
+                d = depth + 4 if v is str and "\n" in v else depth + 3
```

Comment:
This special casing based on the value feels like we're leaking implementation details of the `yv` function out to the caller.  Indentation of the `v` should be handled by `yv` itself based on the provided depth.

The 3 and 4 constants here are a bit unexpected to me; this sort of pattern of passing down the depth usually indicates the actual call depth and not some externally adjusted depth.

Reply:
Nicely spotted :) This was one line that both took me the longest to figure out, and I wasn't really happy with as well. I had another look, and it turned out the special casing is not necessary here (I think it was at one point during the refactoring).